src/retry/mempool.rs

```diff
@@ -0,0 +1,102 @@
+use super::validate::KakarotTransactionValidator;
+use crate::eth_provider::provider::EthDataProvider;
+use reth_transaction_pool::{
+    blobstore::NoopBlobStore, CoinbaseTipOrdering, EthPooledTransaction, Pool, TransactionPool,
+};
+use serde_json::Value;
+use starknet::core::types::Felt;
+use std::{collections::HashSet, fs::File, io::Read, time::Duration};
+use tokio::runtime::Handle;
+
+/// A type alias for the Kakarot Transaction Validator.
+/// Uses the Reth implementation [`TransactionValidationTaskExecutor`].
+pub type Validator<Client> = KakarotTransactionValidator<Client, EthPooledTransaction>;
+
+/// A type alias for the Kakarot Transaction Ordering.
+/// Uses the Reth implementation [`CoinbaseTipOrdering`].
+pub type TransactionOrdering = CoinbaseTipOrdering<EthPooledTransaction>;
+
+/// A type alias for the Kakarot Sequencer Mempool.
+pub type KakarotPool<Client> = Pool<Validator<Client>, TransactionOrdering, NoopBlobStore>;
+
+#[derive(Debug, Default)]
+pub struct AccountManager {
+    accounts: HashSet<Felt>,
+}
+
+impl AccountManager {
+    pub fn new() -> Self {
+        let mut accounts = HashSet::new();
+
+        let mut file = File::open("src/retry/accounts.json").unwrap();
+        let mut contents = String::new();
+        file.read_to_string(&mut contents).unwrap();
+
+        let json: Value = serde_json::from_str(&contents).unwrap();
+        if let Some(array) = json.as_array() {
+            for item in array {
+                if let Some(address) = item.as_str() {
+                    accounts.insert(Felt::from_hex_unchecked(address));
+                }
+            }
+        }
+
+        Self { accounts }
+    }
+
+    pub fn start<SP>(&self, rt_handle: &Handle, eth_provider: &'static EthDataProvider<SP>)
+    where
+        SP: starknet::providers::Provider + Send + Sync + 'static,
+    {
+        let accounts = self.accounts.clone();
+
+        rt_handle.spawn(async move {
+            loop {
+                for address in &accounts {
+                    Self::process_account(address, eth_provider);
+                }
+
+                tokio::time::sleep(Duration::from_secs(1)).await;
+            }
+        });
+    }
+
+    fn process_account<SP>(address: &Felt, eth_provider: &EthDataProvider<SP>)
+    where
+        SP: starknet::providers::Provider + Send + Sync + 'static,
+    {
+        let balance = Self::check_balance(address);
```

Comment:
so this is just a temp that pulls the best transactions and removes them? I'm curious how this can work, won't we get race conditions when a transaction is added from the `send_raw_transaction` and we also access the `best_transactions` here?

Reply:
Yes, I am confused with it myself because there is a problem that is not solved.
1. In the main file we create a provider.
2. Then we create a mempool associated with the provider.
3. Then we associate the mempool with the provider, we put it in it and we give the provider permission to fill it.
4. But at the same time we should pass this provider as a reference to the account manager service so that it has the right to take transactions in it and also to be able to modify the mempool... And there we have a problem if only with the Rust borrow checker because we don't really have the right to do that... So I don't really know.